ci/code_checks.sh

```diff
@@ -0,0 +1,131 @@
+
+    # Check for imports from pandas.core.common instead of `import pandas.core.common as com`
+    MSG='Check for non-standard imports' ; echo $MSG
+    ! grep -R --include="*.py*" -E "from pandas.core.common import " pandas
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for pytest warns' ; echo $MSG
+    ! grep -r -E --include '*.py' 'pytest\.warns' pandas/tests/
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    # Check for the following code in testing: `np.testing` and `np.array_equal`
+    MSG='Check for invalid testing' ; echo $MSG
+    ! grep -r -E --include '*.py' --exclude testing.py '(numpy|np)(\.testing|\.array_equal)' pandas/tests/
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    # Check for the following code in the extension array base tests: `tm.assert_frame_equal` and `tm.assert_series_equal`
+    MSG='Check for invalid EA testing' ; echo $MSG
+    ! grep -r -E --include '*.py' --exclude base.py 'tm.assert_(series|frame)_equal' pandas/tests/extension/base
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for deprecated messages without sphinx directive' ; echo $MSG
+    ! grep -R --include="*.py" --include="*.pyx" -E "(DEPRECATED|DEPRECATE|Deprecated)(:|,|\.)" pandas
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for old-style classes' ; echo $MSG
+    ! grep -R --include="*.py" -E "class\s\S*[^)]:" pandas scripts
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for backticks incorrectly rendering because of missing spaces' ; echo $MSG
+    ! grep -R --include="*.rst" -E "[a-zA-Z0-9]\`\`?[a-zA-Z0-9]" doc/source/
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for incorrect sphinx directives' ; echo $MSG
+    ! grep -R --include="*.py" --include="*.pyx" --include="*.rst" -E "\.\. (autosummary|contents|currentmodule|deprecated|function|image|important|include|ipython|literalinclude|math|module|note|raw|seealso|toctree|versionadded|versionchanged|warning):[^:]" ./pandas ./doc/source
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for modules that pandas should not import' ; echo $MSG
+    python -c "
+import sys
+import pandas
+
+blacklist = {'bs4', 'gcsfs', 'html5lib', 'ipython', 'jinja2' 'hypothesis',
+             'lxml', 'numexpr', 'openpyxl', 'py', 'pytest', 's3fs', 'scipy',
+             'tables', 'xlrd', 'xlsxwriter', 'xlwt'}
+mods = blacklist & set(m.split('.')[0] for m in sys.modules)
+if mods:
+    sys.stderr.write('pandas should not import: {}\n'.format(', '.join(mods)))
+    sys.exit(len(mods))
+    "
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+fi
+
+### DOCTESTS ###
+if [[ -z "$CHECK" || "$CHECK" == "doctests" ]]; then
+
+    MSG='Doctests frame.py' ; echo $MSG
+    pytest --doctest-modules -v pandas/core/frame.py \
+        -k"-axes -combine -itertuples -join -nlargest -nsmallest -nunique -pivot_table -quantile -query -reindex -reindex_axis -replace -round -set_index -stack -to_dict -to_stata"
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
```

Comment:
I didn't see a response on splitting this file up into 3 separate files, you can make a sub-dir fi that is helpful and have a top-level script to call them.

Reply:
I think I mentioned above. What I did is make the script able to run for a specific section. That keeps the advantage of simplicity, and addresses your concerns of being able to run just part of it in `Makefile` or manually. These are the calls:
- `./ci/code_checks.sh lint` <- flake8 and cpplint
- `./ci/code_checks.sh patterns` <- greps
- `./ci/code_checks.sh doctests` <- doctests